.vscode/settings.json

```diff
@@ -13,6 +13,7 @@
     "editor.tabSize": 4
   },
   "editor.rulers": [
-    79
-  ]
+    80,
+    100
```

Comment:
I think this is one PEP convention that is a little ridiculous. Many pythonistas agree. E.g. see [The Best of the Best Practices Guide for Python](https://gist.github.com/sloria/7001839).

Fair enough on 79 vs 80. But I basically figured having a ruler for 79/80 to indicate PEP compliance and a ruler for 100 to indicate a more reasonable max line length would be a better combination.

72 is just flat our ridiculous. It severely limits code readability. Fewer and fewer machines are limited to an 80 char line length and, generally, more (unnecessary) wrapping means more unreadble code. ESPECIALLY for python, for which descriptive variable names are encouraged and indentation is so embedded in the language.

Reply:
Also note the following in PEP:

> "Some teams strongly prefer a longer line length. For code maintained exclusively or primarily by a team that can reach agreement on this issue, it is okay to increase the line length limit up to 99 characters, provided that comments and docstrings are still wrapped at 72 characters."

Reply:
> "Some teams strongly prefer a longer line length. For code maintained exclusively or primarily by a team that can reach agreement on this issue, it is okay to increase the line length limit up to 99 characters, provided that comments and docstrings are still wrapped at 72 characters."

Interesting. Never read that bit. I'm certainly willing to be swayed. ... I do often really like having the four side by side screens though... much more than 79 characters and I do lose that flexibility...

Reply:
Do your "screens" have to be equal in width (i.e. do they all contain text editors)?

I tend to have one text editor next to Cmder, split across one screen so that Cmder gets less than half of the screen and VS Code gets more than half. This is particularly easy in Windows 10, which can handle unequal window snaps

Reply:
80 chars doesn't need anywhere near a half screen. Assuming you're running with 16:9 and a reasonable font size...

Reply:
Two screens at full 1080p:

![image](https://user-images.githubusercontent.com/6559099/56175623-4d6f7d80-603b-11e9-893a-571348a802a2.png)

... I really like the 79 character limit as it allows me to do the above ...

Reply:
The side by side diff case is fair. I do squeeze in with 99 chars, but I lose the file preview in the scrollbar. And the sidebar is a little narrow.

![image](https://user-images.githubusercontent.com/19767103/56176194-7262f100-6039-11e9-8a22-45dea44b37e8.png)

Reply:
There's still the excessive scrolling point. Not to mention many examples of contiguous text that you really don't want to wrap (like URLs) simply not fitting in 72/79, especially with indenting.

Reply:
As a compromise, how about the following guideline from [BigchainDB](http://docs.bigchaindb.com/projects/contributing/en/latest/cross-project-policies/python-style-guide.html)?

> **Maximum Line Length**
> PEP8 has some maximum line length guidelines, starting with “Limit all lines to a maximum of 79 characters” but “for flowing long blocks of text with fewer structural restrictions (docstrings or comments), the line length should be limited to 72 characters.”
> 
> We discussed this at length, and it seems that the consensus is: try to keep line lengths less than 79/72 characters, unless you have a special situation where longer lines would improve readability. (The basic reason is that 79/72 works for everyone, and BigchainDB is an open source project.) As a hard limit, keep all lines less than 119 characters (which is the width of GitHub code review).

Reply:
I've actually even found 99 to be restrictive. So perhaps 119 to be used in extreme cases only (like URLs), since it matches GitHub? ANd I will endeavour to stick to 72/79 from now on?

Reply:
> I've actually even found 99 to be restrictive. So perhaps 119 to be used in extreme cases only (like URLs), since it matches GitHub? And I will endeavour to stick to 72/79 from now on?

Seems like a good compromise. Worthwhile having rulers at 72, 79, and 119.